storage/innobase/buf/buf0dump.cc

```diff
@@ -180,7 +180,7 @@ static void buf_dump_generate_path(char *path, size_t path_size)
 	char	buf[FN_REFLEN];
 
 	mysql_mutex_lock(&LOCK_global_system_variables);
-	snprintf(buf, sizeof buf, "%s/%s", get_buf_dump_dir(),
+	snprintf(buf, sizeof buf, "%s%c%s", get_buf_dump_dir(), FN_LIBCHAR,
 		 srv_buf_dump_filename);
```

Comment:
The following should not induce any run-time overhead:
```c++
	snprintf(buf, sizeof buf, "%s" FN_ROOTDIR "%s", get_buf_dump_dir(),
		 srv_buf_dump_filename);
```
I think that we should adjust also the `format = '%s/%s'` later in the function. That path name is being passed to `buf_dump_status()` where it would be visible to the user.